server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/SnapshotInProgressAllocationDecider.java

```diff
@@ -101,6 +101,16 @@ private static Decision canMove(ShardRouting shardRouting, RoutingAllocation all
                     continue;
                 }
 
+                if (shardSnapshotStatus.state() == SnapshotsInProgress.ShardState.PAUSED_FOR_NODE_REMOVAL
+                    && snapshotsInProgress.isNodeIdForRemoval(shardRouting.currentNodeId())) {
+                    // this shard snapshot is paused pending the removal of its assigned node
+                    //
+                    // NB we check isNodeIdForRemoval() too because if the node was marked for removal and then that mark was removed then
+                    // the shard can still be PAUSED_FOR_NODE_REMOVAL while there are other shards on the node which haven't finished
+                    // pausing yet. In that case the shard is about to go back into INIT state again so we should keep it where it is.
```

Comment:
In my head, the code does the opposite of the comment, but I might be reading it wrongly.

If we wanted the shard to remain throttled, i.e., unable to move off the node when paused but no longer being shutdown, I think we need to check that actual shutdown indications, not the one in `snapshotsInProgress`.

I do not think it is very important though, it is mostly a small race with slightly inoptimal behavior more than anything too important. Anyways, my preference would be to check the actual shutdowns here, unless you have concerns on that?

Reply:
Hm yeah I also think it's backwards now that I re-read it. Not sure what my intentions were here, but yes I now think we should check the actual shutdowns, see 7b285ebc0fd6b341dfa3d5ce582426e01ae23d8f.